### PR TITLE
[Marksmanship] Ullrs Feather Snowshoes Trueshot reduction module added 

### PR DESCRIPTION
**Pull request by Pewtro**
Adds a tracker and effectiveness module for Ullr's Feather Snowshoes if the hunter has these equipped

### CODE REVIEW

File: src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  effectiveTrueshotReductionMs = 0;
+  wastedTrueshotReductionMs = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFeet(ITEMS.ULLRS_FEATHER_SNOWSHOES.id);
+  }
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.AIMED_SHOT.id && spellId !== SPELLS.MARKED_SHOT.id && spellId !== SPELLS.ARCANE_SHOT.id && spellId !== SPELLS.MULTISHOT.id) {
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js]: Although it's quite clear what this condition does, it's uncomfortably long. What about this?
```jsx
// before class
const AFFECTED_ABILITIES = [
   SPELLS.AIMED_SHOT.id,
   SPELLS.MARKED_SHOT.id,
   SPELLS.ARCANE_SHOT.id,
   SPELLS.MULTISHOT.id,
];
.
.
.
if (AFFECTED_ABILITIES.every(id => spellId !== id)) {
...
}

```

File: src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  }
+
+  item() {
+    return {
+      item: ITEMS.ULLRS_FEATHER_SNOWSHOES,
+      result: (
+        <dfn data-tip={`You wasted ${formatNumber(this.wastedTrueshotReductionMs / 1000)} seconds of CDR.<br/> `}>
+          You effectively reduced <SpellLink id={SPELLS.TRUESHOT.id} />s cooldown by a total of {formatNumber(this.effectiveTrueshotReductionMs / 1000)} seconds over the course of the fight.
+        </dfn>
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js]: Although I do like the text, I think it's a tad bit long for an `item()`. Does it fit on one line or how does it look? Maybe a simple "Trueshot CD reduced by X seconds" would be enough.

File: src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      this.wastedTrueshotReductionMs += COOLDOWN_REDUCTION_MS;
+    }
+  }
+
+  item() {
+    return {
+      item: ITEMS.ULLRS_FEATHER_SNOWSHOES,
+      result: (
+        <dfn data-tip={`You wasted ${formatNumber(this.wastedTrueshotReductionMs / 1000)} seconds of CDR.<br/> `}>
+          You effectively reduced <SpellLink id={SPELLS.TRUESHOT.id} /> CD reduced by {formatNumber(this.effectiveTrueshotReductionMs / 1000)} in total.
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js]: Little more wording here - the "reduced" after CD is redundant ("You effectively reduced Trueshot CD **reduced** by ...")

Comment (Pewtro) [reply src/Parser/MarksmanshipHunter/Modules/Items/UllrsFeatherSnowshoes.js]: oh, doing things in between pulls not ideal

**Comment by Pewtro** [comment]
Condensed it down to look like this: 
image


Now it looks like "the item reduced ts by X in total" and that fits in one page atleast

**Pull request merged**